colossalai/auto_parallel/solver/strategy/matmul_strategy_generator.py

```diff
@@ -0,0 +1,312 @@
+        }
+
+        communication_actions = {
+            self.operation_data_mapping['output']: [
+                CommunicationAction(type=CollectiveCommPattern.REDUCE_FWD_IDENTITY_BWD, mesh_dim=mesh_dim_0)
+            ],
+            self.operation_data_mapping['input']: [
+                CommunicationAction(type=CollectiveCommPattern.IDENTITY_FWD_ALLREDUCE_BWD, mesh_dim=mesh_dim_1)
+            ]
+        }
+        return ShardingStrategy_V2(name=name,
+                                   dim_partition_dict=dim_partition_dict,
+                                   communication_actions=communication_actions)
+
+    def recompute_split_both_contract(self, mesh_dim):
+        name = f'RR = RS{mesh_dim} x S{mesh_dim}R'
+        dim_partition_dict = {
+            "input": {
+                -1: [mesh_dim]
+            },
+            self.operation_data_mapping['other']: {
+                self.dim_p: [mesh_dim]
+            },
+            self.operation_data_mapping['bias']: {},
+            self.operation_data_mapping['output']: {},
+        }
+        communication_actions = {
+            self.operation_data_mapping['output']: [
+                CommunicationAction(type=CollectiveCommPattern.REDUCE_FWD_IDENTITY_BWD, mesh_dim=mesh_dim)
+            ],
+        }
+        return ShardingStrategy_V2(name=name,
+                                   dim_partition_dict=dim_partition_dict,
+                                   communication_actions=communication_actions)
+
+    def split_rhs_space_only(self, mesh_dim):
+        name = f'RS{mesh_dim} = RR x RS{mesh_dim}'
+        dim_partition_dict = {
+            "input": {},
+            self.operation_data_mapping['other']: {
+                self.dim_q: [mesh_dim]
+            },
+            self.operation_data_mapping['bias']: {
+                -1: [mesh_dim]
+            },
+            self.operation_data_mapping['output']: {
+                -1: [mesh_dim]
+            },
+        }
+        communication_actions = {
+            self.operation_data_mapping['input']: [
+                CommunicationAction(type=CollectiveCommPattern.IDENTITY_FWD_ALLREDUCE_BWD, mesh_dim=mesh_dim)
+            ],
+        }
+        return ShardingStrategy_V2(name=name,
+                                   dim_partition_dict=dim_partition_dict,
+                                   communication_actions=communication_actions)
+
+    def split_lhs_1st_dim_1d(self, mesh_dim_0, mesh_dim_1):
+        name = f'S{mesh_dim_0}{mesh_dim_1}R = S{mesh_dim_0}{mesh_dim_1}R x RR'
+        dim_partition_dict = {
+            "input": {
+                0: [mesh_dim_0, mesh_dim_1]
+            },
+            self.operation_data_mapping['other']: {},
+            self.operation_data_mapping['bias']: {},
+            self.operation_data_mapping['output']: {
+                0: [mesh_dim_0, mesh_dim_1]
+            },
+        }
+        communication_actions = {
+            self.operation_data_mapping['other']: [
```

Comment:
For better performance, flatten device mesh should be used here.

Reply:
updated with comm spec.